crates/evm/traces/src/identifier/etherscan.rs

```diff
@@ -112,11 +113,30 @@ impl TraceIdentifier for EtherscanIdentifier {
         );
 
         for (addr, _) in addresses {
-            if !self.contracts.contains_key(addr) {
+            if self.contracts.contains_key(addr) {
+                resolved.push(*addr);
+            } else {
                 fetcher.push(*addr);
             }
         }
 
+        let resolved = resolved
+            .into_iter()
+            .map(|addr| {
+                let metadata = self.contracts.get(&addr).unwrap();
+                let label = metadata.contract_name.clone();
+                let abi = metadata.abi().ok().map(Cow::Owned);
+
+                AddressIdentity {
+                    address: addr,
+                    label: Some(label.clone()),
+                    contract: Some(label),
+                    abi,
+                    artifact_id: None,
+                }
+            })
+            .collect::<Vec<AddressIdentity<'_>>>();
```

Comment:
this can be done in the for loop above?

so that we track `AddressIdentity` in resolved?

Reply:
Good point, updated